oeps/architectural-decisions/oep-XXXX-micro-frontend-domains.rst

```diff
@@ -0,0 +1,171 @@
+================================
+OEP-XXXX: Micro-frontend Domains
+================================
+
+.. list-table::
+   :widths: 25 75
+
+   * - OEP
+     - :doc:`OEP-XXXX <oep-XXXX-micro-frontend-domains>`
+   * - Title
+     - Micro-frontend Domains
+   * - Last Modified
+     - 2022-11-09
+   * - Authors
+     - Adolfo R. Brandes <adolfo@tcril.org>
+   * - Arbiter
+     - TBD
+   * - Status
+     - Draft
+   * - Type
+     - Architecture
+   * - Created
+     - 2022-11-09
+   * - Review Period
+     - TBD
+   * - Resolution
+     - TBD
+   * - References
+     - TBD
+
+
+********
+Abstract
+********
+
+The decision to gradually implement a micro-frontend architecture for Open edX was a largely successful attempt at avoiding some limitations of the monolithic ``edx-platform`` codebase.  In particular, it allowed the project to integrate more and better features by leveraging newer frontend technology, all much faster than would have been possible otherwise.  At the same time, because features encapsulated in micro-frontends (henceforth, "MFEs") are very loosely coupled to each other, this architecture eliminates most integration bottlenecks, allowing individual teams working on any given feature to not be a lot more *productive*.
+
+However, roughly 3 years into this gradual re-platforming, several drawbacks to the chosen implentation of complete single-page app ("SPA") independence have been identified.  This OEP proposes a middle-ground based on Domain Driven Development ("DDD") and current industry best-practices, where a minor increase in integration costs is traded for large potential gains in user experience, customizability, ease of deployment, and in certain cases, even development efficiency.
+
+
+**********
+Motivation
+**********
+
+When moving to micro-frontends, a (now known to be) uncommon choice was made: to have *completely* independent SPAs, each with their separate deployment, endpoint, and different technology stack.  While the goal of developing platform features more efficiently was achieved, the price to pay was not insignificant.  Here are some critical issues that the new architecture introduced:
```

Comment:
nit: what is meant by "endpoint" here? Is this referring to the URL subdomain where the MFE is hosted?

Reply:
It is a bad attempt at a catch-all term for individual MFEs having different apparent domains/ports/URLs even though they're supposed to be perceived as being part of the same app, and also the fact they are potentially hosted separately.  I'll think of something better to say, here.